tests/keras/backend/backend_test.py

```diff
@@ -262,6 +262,53 @@ def ref_separable_conv(x, w1, w2, padding, data_format):
     return ref_conv(x2, w2, padding, data_format)
 
 
+def ref_pool(x, pool_size, strides, padding, data_format, pool_mode):
+    if data_format == 'channels_last':
+        if x.ndim == 3:
+            x = np.transpose(x, (0, 2, 1))
+        elif x.ndim == 4:
+            x = np.transpose(x, (0, 3, 1, 2))
+        else:
+            x = np.transpose(x, (0, 4, 1, 2, 3))
+
+    if padding == 'same':
+        pad = [(0, 0), (0, 0)] + [(s // 2, s // 2) for s in pool_size]
+        x = np.pad(x, pad, 'constant', constant_values=-np.inf)
+
+    y = []
+    for i in range(x.shape[0]):
+        _y = []
+        for j in range(x.shape[1]):
+            __y = []
```

Comment:
The notation seems unusual (`y`, `_y`, `__y`). Please find something more descriptive and conventional